lib/membrane_rtc_engine/engine.ex

```diff
@@ -869,17 +870,23 @@ defmodule Membrane.RTC.Engine do
     {:ok, state}
   end
 
-  defp handle_tee_notification({:encoding_switched, endpoint_id, encoding}, track_id, state) do
-    # send event that endpoint with id `from_endpoint_id` is sending encoding `encoding` for track
-    # `track_id` now
+  defp validate_track(track) do
+    variants = MapSet.new(track.variants)
+    supported_variants = Track.supported_variants() |> MapSet.new()
 
-    {from_endpoint_id, _endpoint} =
-      Enum.find(state.endpoints, fn {_, endpoint} ->
-        Endpoint.get_track_by_id(endpoint, track_id) != nil
-      end)
+    cond do
+      variants == MapSet.new([]) ->
+        raise PublishTrackError, track: track
 
-    dispatch(endpoint_id, MediaEvent.encoding_switched(from_endpoint_id, track_id, encoding))
-    {:ok, state}
+      track.type == :audio and not MapSet.equal?(variants, MapSet.new([:high])) ->
+        raise PublishTrackError, track: track
+
+      MapSet.subset?(variants, supported_variants) == false ->
+        raise PublishTrackError, track: track
+
+      true ->
+        :ok
+    end
```

Comment:
Isn't it a function which should be stored in `Track.ex`? Also I would change name of this function to maybe `validate_track_variants`?

Reply:
This could be in `Track.new()` however someone can modify Track struct on its own, besides Track.new, which will result in publishing incorrect track :thinking:

Reply:
I was more thinking that in `Track.ex` we would have the function `validate_track_variants`, which would return `:ok | :error`. This way we could check variants in any module it could be needed.

Reply:
On the one side, that's true but on the other side such approach makes the implementation a little bit more complex as I would have to create function with the following definition

```
@spec validate(struct) :: :ok | {:error, :invalid_variants | :invalid_xxx}
```

Then in engine we would have to interpret each error tuple and raise different error. Could we keep the current version until we need this feature in other places too? When there is a new use case it will be easier to move this function to `Track` module.